src/ppl/src/IOPlacer.cpp

```diff
@@ -693,6 +693,10 @@ void IOPlacer::writePinPlacement(const char* file_name)
 
   std::ofstream out(filename);
 
+  if (!out.is_open()) {
```

Comment:
it is simpler and more robust to just use "if (!out)" as that also checks for a good state.